src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -291,6 +291,41 @@ public int CompareTo(object? value)
 
         public static TimeSpan FromDays(double value) => Interval(value, TicksPerDay);
 
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> that represents a specified number of days.
+        /// </summary>
+        /// <param name="days">A number of days.</param>
+        /// <returns>An object that represents <paramref name="days"/>.</returns>
+        /// <exception cref="OverflowException">
+        /// <paramref name="days"/> is less than <see cref="MinValue"/> or greater than <see cref="MaxValue"/>.
+        /// </exception>
+        public static TimeSpan FromDays(int days)
+        {
+            return FromMicroseconds(days * MicrosecondsPerDay);
```

Comment:
>days * MicrosecondsPerDay [](http://example.com/codeflow?start=36&length=25)

This multiplication by itself can overflow and make the operation succussed without exception I guess. 

For example, using `1067519900` days will results to `-1008547758080` microseconds which is going to be considered as a valid value while it is really overflowed. 
Maybe the best here is to check the days values and directly convert the result to ticks if it is valid.

This apply to similar calling of `From` operations with other units (hours, minutes...etc.)

CC @tannergooding

Reply:
Should we validate that it's less than `MaxDays` (`10_675_199`)? `1_067_519_900` is much larger

Reply:
Yes, that is what I was suggesting when I said `the best thing that can be done here is to check the day's value and directly convert the result to ticks if it is valid. `